src/daemon/dlt_daemon_socket.c

```diff
@@ -145,27 +145,81 @@ int dlt_daemon_socket_close(int sock)
 int dlt_daemon_socket_send(int sock,void* data1,int size1,void* data2,int size2,char serialheader)
 {
     /* Optional: Send serial header, if requested */
+    int ret = DLT_RETURN_OK;
     if (serialheader)
     {
-        if ( 0 > send(sock, dltSerialHeader,sizeof(dltSerialHeader),0) )
-            return DLT_DAEMON_ERROR_SEND_FAILED;
-
+         ret = dlt_daemon_socket_sendreliable(sock, (void *) dltSerialHeader, sizeof(dltSerialHeader));
+         if(ret != DLT_RETURN_OK)
+             return ret;
     }
 
     /* Send data */
-    if(data1 && size1>0)
+    if(data1 != NULL && size1 > 0)
     {
-        if (0 > send(sock, data1,size1,0))
-            return DLT_DAEMON_ERROR_SEND_FAILED;
+        ret = dlt_daemon_socket_sendreliable(sock, data1, size1);
+        if(ret != DLT_RETURN_OK)
+            return ret;
     }
 
-    if(data2 && size2>0)
+    if(data2 != NULL && size2 > 0)
     {
-        if (0 > send(sock, data2,size2,0))
-        return DLT_DAEMON_ERROR_SEND_FAILED;
+        ret = dlt_daemon_socket_sendreliable(sock, data2, size2);
+        if(ret != DLT_RETURN_OK)
+            return ret;
     }
 
-    return DLT_DAEMON_ERROR_OK;
+    return ret;
+}
+
+int dlt_daemon_socket_sendreliable(int sock, void* buffer,int  message_size)
+{        
+   int remaining_size = message_size;
+   int data_sent = 0;
+   int errno_send = 0;
+   int resend_ou = 1;
+
+   //try to resend data, if it was not sent for the first time (max retries: DLT_DAEMON_SOCKET_MAXRESEND_ATTEMPTS)
+   //this implementation prevents occasional corrupted messages, when there is e.g. system send buffer full, high system utilization, etc.
+   while (remaining_size > 0 && resend_ou <= DLT_DAEMON_SOCKET_MAXRESEND_ATTEMPTS)
+   {
+       int resend_in = 0;
+       ssize_t ret = 0;
+
+       //repeat due to some temporal issues
+       do
+       {
+            ret = send(sock, buffer + data_sent, remaining_size, 0);
+            if((ret < 0) && (errno != EINTR && errno != EAGAIN && errno != EWOULDBLOCK))
```

Comment:
Condition in line no:192 and 199 contradicts. retry print was printed if errno is none of the mentioned errors but actually retry was happening only on any of these errors

Reply:
I agree, the message should be printed in case of given conditions occurred. 
Must be updated.